test/meson.build

```diff
@@ -1,23 +1,23 @@
 nntrainer_test_inc = include_directories('./include')
 
 nntrainer_test_deps = [
-  nntrainer_dep #gtest is linked in nntrainer_testutil_lib
+  nntrainer_dep
 ]
 
-# build test util when gtest is found
-if gtest_dep.found()
-  nntrainer_testutil_lib = static_library(
-    'nntrainer_test_util',
-    'nntrainer_test_util.cpp',
-    dependencies: [nntrainer_test_deps, gtest_dep],
-    include_directories: nntrainer_test_inc
-  )
-  nntrainer_testutil_dep = declare_dependency(
-    link_with: nntrainer_testutil_lib,
-    include_directories: nntrainer_test_inc
-  )
-  nntrainer_test_deps += nntrainer_testutil_dep
-endif
+nntrainer_testutil_lib = static_library(
+  'nntrainer_test_util',
+  'nntrainer_test_util.cpp',
+  dependencies: [nntrainer_test_deps],
+  include_directories: nntrainer_test_inc
+)
+
+nntrainer_testutil_dep = declare_dependency(
+  link_with: nntrainer_testutil_lib,
+  include_directories: nntrainer_test_inc
+)
+
+nntrainer_test_deps += nntrainer_testutil_dep
+nntrainer_test_deps += gtest_dep
```

Comment:
while you are editing around gtest, can you disable unit tests build if gtest_dep not found.

Reply:
Added a check if gtest not found, make error when test is enabled